src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -908,16 +908,17 @@ namespace
                     precomputed_mis_weights = scattering_coef;
                 else
                     Spectrum::upgrade(scattering_coef, precomputed_mis_weights);
-                precomputed_mis_weights *= vertex.m_throughput;
+                if (vertex.m_throughput.size() <= channel_count)
```

Comment:
I'm not sure to understand this test, in fact it looks wrong.

There are really only two cases when working with spectra: either they are the same size (both RGB or both spectral), or they're not in which case the RGB one must be upgraded.